pkg/agent/knapsack/knapsack.go

```diff
@@ -75,6 +75,10 @@ func (k *knapsack) BboltDB() *bbolt.DB {
 
 // Stores interface methods
 
+func (k *knapsack) Stores() map[storage.Store]types.KVStore {
```

Comment:
You mentioned in a different comment `exposing all the raw stores` -- I think easier to address here in one thread. I don't think this really exposes much more than is already exposed? Every raw store is already exposed as its `knapsack.StoreName()` method -- this just collects them all into one list.

Reply:
There are two slightly related conversations.

One is where we define the `WipeDb` logic. The other is the method of implementation.

For implementation, I think we cannot remove the DB. It's already open, and closing, deleting, recreating sounds impossible. I was suggesting we use the bbolt method `DeleteBucket`, instead of iterating the keys. I suspect it's faster. 

In terms of where it lives, it feels a little cleaner to me to have `knapsack.WipeDb()` than to keep the logic here.

Reply:
I'm not sure that feels cleaner to me -- I prefer exposing `Stores` in knapsack over making a `WipeDb` method available to anywhere that uses the knapsack.

Reply:
If you expose Store, any caller can implement the same.

I always imagined we'd grow to _not_ declaring knapsack in the signature, and instead the specific methods we need. But we haven't gotten there yet.

Anyhow, eventually I want to get away from exposing Stores at all. We can't move away from boltdb when things use it.

Reply:
Aren't the stores an interface so that we can swap away from bbolt seamlessly in the future? I thought it was direct usage of bbolt that was an issue.

Reply:
Oh, you are correct! Stores is the KV store. I guess that does feel better. 

Okay, if you want to use those, we should make a Destroy or DeleteAll, or similar and not iterate

Reply:
Done! Added a `DeleteAll`